src/main/java/com/google/devtools/build/lib/analysis/constraints/IncompatibleTargetChecker.java

```diff
@@ -148,8 +152,16 @@ public StateMachine step(Tasks tasks, ExtendedEventHandler listener) {
         return DONE;
       }
 
-      // Resolves the constraint labels.
-      for (Label label : attrs.get("target_compatible_with", BuildType.LABEL_LIST)) {
+      // Resolves the constraint labels, checking for invalid configured attributes.
+      List<Label> targetCompatibleWith;
+      try {
+        targetCompatibleWith = attrs.getAndValidate("target_compatible_with", BuildType.LABEL_LIST);
+      } catch (ValidationException e) {
+        sink.accept(Optional.empty());
```

Comment:
Consider setting just ValidationException here.

This class is a bit of a special case because it doesn't have real output when validation passes so a comment on ResultSink could help readers.